addons/analytic/tests/test_analytic_account.py

```diff
@@ -186,3 +186,24 @@ def test_analytic_plan_account_child(self):
         plans_json = self.env['account.analytic.plan'].get_relevant_plans()
         self.assertEqual(2, len(plans_json) - self.analytic_plan_offset,
                          "The parent plan should be available even if the analytic account is set on child of third generation")
+
+    def test_guva(self):
+        self.analytic_plan = self.env['account.analytic.plan'].create({
+            'name': 'Parent Plan',
+        })
+        self.analytic_sub_plan = self.env['account.analytic.plan'].create({
+            'name': 'Sub Plan',
+            'parent_id': self.analytic_plan.id,
+        })
+        self.analytic_sub_sub_plan = self.env['account.analytic.plan'].create({
+            'name': 'Sub Sub Plan',
+            'parent_id': self.analytic_sub_plan.id,
+        })
+        self.env['account.analytic.account'].create({'name': 'Account', 'plan_id': self.analytic_plan.id})
```

Comment:
You can batch the accounts creation into one call.
Can't do the same for the plans as you need the parent each time